types/koa-ratelimit/index.d.ts

```diff
@@ -80,6 +80,16 @@ declare namespace KoaRatelimit {
          * A relation of header to the header's display name.
          */
         headers?: HeaderNameOptions;
+
+        /**
+         * If function returns true, middleware exits before limiting
+         */
+        whitelist?: (context: Context) => Promise<boolean>;
```

Comment:
Consider using a union return type to also support synchronous functions: `boolean | Promise<boolean>`